sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp

```diff
@@ -24,6 +24,30 @@ namespace Azure { namespace Core { namespace Http {
     constexpr std::chrono::milliseconds DefaultConnectionTimeout = std::chrono::minutes(5);
   } // namespace _detail
 
+  /**
+   * @brief A #TransportException which includes the CURL error code which triggered the exception
+   * (if any).
+   *
+   */
+  struct CurlTransportException : public TransportException
```

Comment:
```suggestion
  struct CurlTransportException final : public TransportException
```

Reply:
Personally, I think the obsession with `final` is going to be a long-term problem, but I'm not going to fight it. But at a minimum, it completely blocks customers ability to mock our functionality.